src/exception.cpp

```diff
@@ -562,18 +509,20 @@ namespace hpx
         std::ostringstream strm;
         strm << "\n";
 
+        // add full build information
+        strm << full_build_string();
+
+        std::string const* env = xi.get<hpx::detail::throw_env>();
+        if (env && !env->empty())
+            strm << "{env}: " << *env;
+
         std::string const* back_trace =
             xi.get<hpx::detail::throw_stacktrace>();
         if (back_trace && !back_trace->empty()) {
             // FIXME: add indentation to stack frame information
             strm << "{stack-trace}: " << *back_trace << "\n";
         }
 
```

Comment:
We've had the stack backtrace first as users where complaining that they can't find it in the output easily...